packages/runtime/src/traits/core/state.tsx

```diff
@@ -1,30 +1,44 @@
-import { useEffect } from 'react';
 import { createTrait } from '@meta-ui/core';
 import { Static, Type } from '@sinclair/typebox';
 import { TraitImplementation } from '../../registry';
+import { stateStore } from '../../store';
+
+const HasInitializedMap = new Map<string, boolean>();
+
+type KeyValue = { key: string; value: unknown };
 
 const useStateTrait: TraitImplementation<{
   key: Static<typeof KeyPropertySchema>;
   initialValue: Static<typeof InitialValuePropertySchema>;
-}> = ({ key, initialValue, mergeState, subscribeMethods }) => {
-  useEffect(() => {
+}> = ({ key, initialValue, componentId, mergeState, subscribeMethods }) => {
+  const hashId = `#${componentId}@${key}`;
+  let hasInitialized = HasInitializedMap.get(hashId);
+
+  if (!hasInitialized) {
     mergeState({ [key]: initialValue });
 
-    subscribeMethods({
-      setValue(value) {
+    const upperCaseKey = capitalizeFirstLetter(key);
```

Comment:
not used in the latest implementation?

Reply:
Yes. I should add a lint rule for unused vairable.